### ISSUE TITLE
Linkit creating node based link paths (2.8.x and later)

### ISSUE THREAD

**Issue by natebyerley** [az-digital/az_quickstart#3125]
## Problem/Motivation
Discovered by Karl Timmermann, partner developer with HR

### Describe the bug

When using Linkit to add a link to text in 2.7 - the link appears on the front end as a path alias: 
[https://test-azs-jallenfaculty.pantheonsite.io/linkit-test](https://test-azs-jallenfaculty.pantheonsite.io/cas?destination=node/52/edit) - this site as of 1.25.2024 is still on 2.7, may be updated to 2.9 in next few weeks

When using Linkit to add a link to text in 2.8 - the link appears on the front end as a node path: 
[https://live-azs-natetest.pantheonsite.io/linkit-test](https://live-azs-natetest.pantheonsite.io/linkit-test) (login/pw both "admin")

Issue persists in 2.9: [https://qs29beta1-azs-quickstart.pantheonsite.io/linkit-test](https://qs29beta1-azs-quickstart.pantheonsite.io/linkit-test)

#### Dana noted: 

There is also no tooltip visible in Linkit.

#### Brian noted:

I was able to confirm this issue using a [Financial Aid multidev on 2.7.1](https://testingpage-sasg-financialaid.pantheonsite.io/en/brian-test-page). With CKEditor 4, <a> elements include some extra data attributes which allow Linkit to replace the /node/xxx href with the URL alias when the page is rendered. With CKEditor 5, these data attributes don’t seem to be added, even when I allow those attributes in the CKEditor settings.

#### Some related Linkit issues:
[Show URL alias after autocomplete selection instead of internal route (e.g., /node/123)](https://www.drupal.org/project/linkit/issues/2877535)
[Linkit rendering strips data-entity-type and data-entity-uuid](https://www.drupal.org/project/linkit/issues/3396049)

## Proposed resolution
Revert to Linkit behavior evident in 2.7
[https://test-azs-jallenfaculty.pantheonsite.io/linkit-test](https://test-azs-jallenfaculty.pantheonsite.io/cas?destination=node/52/edit)

**Comment by bberndt-uaz** [comment]
To resolve Quickstart issue #2869, we added a patch from this Linkit issue: ["This link has no URL" when inserting text before or after a link](https://www.drupal.org/project/linkit/issues/3364963). Unfortunately, it seems this patch is resulting in the data attributes not being added to `<a>` elements in the editor. I confirmed that removing the patch allowed the URL aliases to work again, but the problem from #2869 also reappeared. I think we'll need a new version of the patch to resolve this issue.
